common/util/rviz_plugins/autoware_auto_perception_rviz_plugin/include/object_detection/object_polygon_detail.hpp

```diff
@@ -0,0 +1,250 @@
+// Copyright 2021 Apex.AI, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+/// \brief This file defines some helper functions used by ObjectPolygonDisplayBase class
+#ifndef OBJECT_DETECTION__OBJECT_POLYGON_DETAIL_HPP_
+#define OBJECT_DETECTION__OBJECT_POLYGON_DETAIL_HPP_
+
+#include <rclcpp/logging.hpp>
+#include <rclcpp/rclcpp.hpp>
+#include <visibility_control.hpp>
+
+#include <autoware_auto_perception_msgs/msg/object_classification.hpp>
+#include <autoware_auto_perception_msgs/msg/predicted_path.hpp>
+#include <autoware_auto_perception_msgs/msg/shape.hpp>
+#include <geometry_msgs/msg/pose_with_covariance.hpp>
+#include <geometry_msgs/msg/twist.hpp>
+#include <geometry_msgs/msg/twist_with_covariance.hpp>
+#include <visualization_msgs/msg/marker.hpp>
+
+#include <map>
+#include <string>
+#include <vector>
+
+namespace autoware
+{
+namespace rviz_plugins
+{
+namespace object_detection
+{
+namespace detail
+{
+// Struct to define all the configurable visual properties of an object of a particular
+// classification type
+struct ObjectPropertyValues
+{
+  // Classified type of the object
+  std::string label;
+  // Color for the type of the object
+  std::array<int, 3> color;
+  // Alpha values for the type of the object
+  float alpha{0.999F};
+};
+
+// Map defining colors according to value of label field in ObjectClassification msg
+const std::map<
+  autoware_auto_perception_msgs::msg::ObjectClassification::_label_type, ObjectPropertyValues>
+  // Color map is based on cityscapes color
+  kDefaultObjectPropertyValues = {
+    {autoware_auto_perception_msgs::msg::ObjectClassification::UNKNOWN,
+     {"UNKNOWN", {255, 255, 255}}},
+    {autoware_auto_perception_msgs::msg::ObjectClassification::CAR, {"CAR", {0, 255, 142}}},
+    {autoware_auto_perception_msgs::msg::ObjectClassification::BUS, {"BUS", {0, 200, 100}}},
+    {autoware_auto_perception_msgs::msg::ObjectClassification::PEDESTRIAN,
+     {"PEDESTRIAN", {220, 20, 60}}},
+    {autoware_auto_perception_msgs::msg::ObjectClassification::BICYCLE, {"CYCLIST", {119, 11, 32}}},
+    {autoware_auto_perception_msgs::msg::ObjectClassification::MOTORCYCLE,
+     {"MOTORCYCLE", {0, 0, 230}}},
+    {autoware_auto_perception_msgs::msg::ObjectClassification::TRAILER, {"TRAILER", {0, 80, 100}}},
+    {autoware_auto_perception_msgs::msg::ObjectClassification::TRUCK, {"TRUCK", {0, 0, 70}}}};
```

Comment:
@yukkysaito 
We can change the polygon color for each label.
If you have color request, please let me know.

Reply:
I may change it later, but can you do this for now?
pedestrian : rgb(255,192,203)
vehicle : rgb(30,144,255)
motorcycle : rgb(255,204,102)

Reply:
I changed colors.

https://github.com/tier4/AutowareArchitectureProposal.iv/pull/574/commits/54c53389cc6d50ead3012eceacddf73e5117d1df
![Screenshot from 2021-11-11 09-07-45](https://user-images.githubusercontent.com/22497720/141213527-fe3323d0-8d5e-4fda-8caf-dfedf94be549.png)

Reply:
Can you make bicycle the same color as motorcycle?

Reply:
https://github.com/tier4/AutowareArchitectureProposal.iv/pull/574/commits/0abb30505e7b668f6be6dc9d6230b8984f6bdf6f